athena-cloudwatch/src/main/java/com/amazonaws/athena/connectors/cloudwatch/CloudwatchRecordHandler.java

```diff
@@ -99,54 +106,74 @@ protected CloudwatchRecordHandler(AmazonS3 amazonS3, AWSSecretsManager secretsMa
      */
     @Override
     protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recordsRequest, QueryStatusChecker queryStatusChecker)
-            throws TimeoutException
+            throws TimeoutException, InterruptedException
     {
-        String continuationToken = null;
-        TableName tableName = recordsRequest.getTableName();
-        Split split = recordsRequest.getSplit();
-        invoker.setBlockSpiller(spiller);
-        do {
-            final String actualContinuationToken = continuationToken;
-            GetLogEventsResult logEventsResult = invoker.invoke(() -> awsLogs.getLogEvents(
-                    pushDownConstraints(recordsRequest.getConstraints(),
-                            new GetLogEventsRequest()
-                                    .withLogGroupName(split.getProperty(LOG_GROUP_FIELD))
-                                    //We use the property instead of the table name because of the special all_streams table
-                                    .withLogStreamName(split.getProperty(LOG_STREAM_FIELD))
-                                    .withNextToken(actualContinuationToken)
-                                    // must be set to use nextToken correctly
-                                    .withStartFromHead(true)
-                    )));
-
-            if (continuationToken == null || !continuationToken.equals(logEventsResult.getNextForwardToken())) {
-                continuationToken = logEventsResult.getNextForwardToken();
-            }
-            else {
-                continuationToken = null;
-            }
+        if (recordsRequest.getConstraints().isQueryPassThrough()) {
+            Map<String, String> qptArguments = recordsRequest.getConstraints().getQueryPassthroughArguments();
+            queryPassthrough.verify(qptArguments);
+            GetQueryResultsResult getQueryResultsResult = getResult(invoker, awsLogs, qptArguments, Integer.parseInt(qptArguments.get(CloudwatchQueryPassthrough.LIMIT)));
```

Comment:
is "limit" a required parameter for QPT request on CW?

Reply:
Also, can we separate QPT read record to another method? to make it easier to read.

Reply:
>is "limit" a required parameter for QPT request on CW?

Can it be null?

Reply:
as we are calling verify, so all args are required. so in this case it will never be null. In one more case, we are taking limit as required arg is to avoid timeout in case of cloudwatch API call.
but in Cloudwatch stsrtQuery API, the limit is optional.

separated the QPT read record to another method.